File: web-ide/frontend/src/types/cell.ts
export type CellOutput = {
  name: 'stdout' | 'stderr' | 'result';  // 출력 타입
  text: string[];
};

export type CellExecutionResult = {
  code: string;
  stdout: string;
  stderr: string;
  result: string | null;
  cell_id: string;
};



export type Cell = {
  cell_type: "code" | "markdown";
  execution_count: number | null;
  id: string;
  metadata: {
    id: string;
    last_run?: string;
  };
  outputs: CellOutput[];
  source: string[];
};

export interface Content {
  cells: Cell[];
}

// export interface IDEFile {
//   id: string;
//   name: string;
//   content: Content;
//   folder_id: number;
//   project_id: number;
//   created_at: string;
//   updated_at: string;
// }

File: web-ide/backend/runtime_python/models/cell.py
# models/cell.py
from pydantic import BaseModel
from enum import Enum
from typing import List

class CellType(Enum):
    CODE = "code"
    MARKDOWN = "markdown"

class CellCreate(BaseModel):
    cell_type: CellType
    source: str = ""  # 기본값을 빈 문자열로 설정)

class CellResponse(BaseModel):
    cell_id: str

class MarkdownUpdate(BaseModel):
    cell_id: str
    source: List[str]

